view/frontend/templates/catalog/product/view/type/grouped/product-protection-offer.phtml

```diff
@@ -7,6 +7,8 @@
 /** @var $block Magento\GroupedProduct\Block\Product\View\Type\Grouped */
 
 $categoryName = '';
+/** encoding to prevent whitespace and other abnormal characters from being used in a div id */
+$encodedProductSku = urlencode($product->getSku());
```

Comment:
i think this logic needs to be moved inside the `foreach` block on line 19 - the way it is now we could get multiple <td> elements with the same `id`